server/data/data-api/src/main/java/org/apache/james/rrt/lib/UserRewritter.java

```diff
@@ -90,7 +90,7 @@ public UserRewritter generateUserRewriter(String mapping) {
                         .map(User::fromUsername);
                 } catch (PatternSyntaxException e) {
                     LOGGER.error("Exception during regexMap processing: ", e);
-                    return Optional.of(User.fromUsername(Mapping.Type.Regex.asPrefix() + mapping));
+                    return Optional.empty();
```

Comment:
I can see a possible regression here: do we have rules that are chained in a way we expect the second rule to be based on a regex of the first rule?

Reply:
Hm... Not sure to really know regarding this... What I do understand though, in my point of view : 
* with the last restrictions on Username, this is likely to fail in most cases as I explained above if the format of the regex is wrong, as basic characters in a regex like `()` are forbidden in a Username now
* is it normal to return something like this if the mapping is wrong in the first place?

Do we really use the regex in James actually by the way?

Reply:
> I can see a possible regression here: do we have rules that are chained in a way we expect the second rule to be based on a regex of the first rule?

1. This is in error handling. As this is error handling, this can't be used for regex chaining.

BTW @Arsnael we should write one cucumber test for regex chaining to prove @rouazana wrong.

2. Saying "a rule is a user" is basically wrong. This is just a broken feature, crossed with meaningless tests.

Reply:
Done